R/navbar.R

```diff
@@ -70,8 +70,12 @@ navbar_components <- function(pkg = ".") {
   }
   menu$news <- navbar_news(pkg)
 
-  if (!is.null(pkg$github_url)) {
-    menu$github <- menu_icon("github", pkg$github_url, style = "fab")
+  if (isTRUE(grepl("github\\.com", repo_home(pkg)))) {
```

Comment:
This is probably mildly premature, but I think it'd be useful to extract this out into something like:

```R
repo_type <- function(pkg) {
  home <- repo_home(pkg)

  if (grepl("github\\.com", home) {
     "github"
  } else if (...) {
    "gitlab"
  } else {
   "other"
  }
}
```

And then you could that function plus a switch statement here.

The `isTRUE()` also makes me slightly nervous because it implies that you think `grepl()` might return something other than a `TRUE` or `FALSE` here and I don't know why. You could make that more clear in `repo_type` by checking the type/length/value of `repo_home()`.